internal/cmd/controller/reconciler/gitrepo_controller.go

```diff
@@ -68,6 +69,8 @@ func (r *GitRepoReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, nil
 	}
 
+	metrics.CollectGitRepoMetrics(gitrepo)
```

Comment:
Newbie question: what happens when a resource, in this case a GitRepo, is deleted? Will its metrics live forever with their last computed values, or will they be garbage-collected at some point?

Reply:
Also, I wonder if we might want to bring metrics collection closer to (concretely right after) resource creation (L139-181), since most gitrepo metrics deal with resource counts.
A downside of this would be that, if resource creation fails for some reason, then metrics would not be refreshed. Would it make sense to compute metrics _more than once_ per reconciliation loop, eg. once here and once after resource creation, just to be on the safe side? Do we want the _observations_total_ metric to reflect a number of reconciliation loops, in which case that would be an issue?